src/buffers/disk.rs

```diff
@@ -227,9 +232,12 @@ impl Drop for Reader {
 
 impl Reader {
     fn delete_acked(&mut self) {
-        let num_to_delete = self.ack_counter.swap(0, Ordering::Relaxed);
+        let num_to_delete = self.ack_counter.load(Ordering::Relaxed);
+        let blocked_writers = self.blocked_write_task_count.load(Ordering::Relaxed);
```

Comment:
why relaxed?

Reply:
Because that's what we're already using. I'm concerned more with the overall perf impact right now than the specifics of the atomics ordering.